fe/fe-core/src/main/java/org/apache/doris/load/loadv2/SparkEtlJobHandler.java

```diff
@@ -140,93 +137,98 @@ public void submitEtlJob(long loadJobId, String loadLabel, EtlJobConfig etlJobCo
+            if (!FeConstants.runningUnitTest) {
+                SparkLauncherMonitors.LogMonitor logMonitor = SparkLauncherMonitors.createLogMonitor(handle);
+                logMonitor.setSubmitTimeoutMs(GET_APPID_TIMEOUT_MS);
+                logMonitor.start();
+                try {
+                    logMonitor.join();
+                } catch (InterruptedException e) {
+                    logMonitor.interrupt();
+                    throw new LoadException(errMsg + e.getMessage());
+                }
+            }
+            appId = handle.getAppId();
+            state = handle.getState();
         } catch (IOException e) {
             LOG.warn(errMsg, e);
             throw new LoadException(errMsg + e.getMessage());
         }
 
-        while (retry++ < GET_APPID_MAX_RETRY_TIMES) {
-            appId = handle.getAppId();
-            if (appId != null) {
-                break;
-            }
-
-            // check state and retry
-            state = handle.getState();
-            if (fromSparkState(state) == TEtlState.CANCELLED) {
-                throw new LoadException(errMsg + "spark app state: " + state.toString());
-            }
-            if (retry >= GET_APPID_MAX_RETRY_TIMES) {
-                throw new LoadException(errMsg + "wait too much time for getting appid. spark app state: "
-                                                + state.toString());
-            }
+        if (fromSparkState(state) == TEtlState.CANCELLED) {
+            throw new LoadException(errMsg + "spark app state: " + state.toString() + ", loadJobId:" + loadJobId);
+        }
 
-            // log
-            if (retry % 10 == 0) {
-                LOG.info("spark appid that handle get is null. load job id: {}, state: {}, retry times: {}",
-                         loadJobId, state.toString(), retry);
-            }
-            try {
-                Thread.sleep(GET_APPID_SLEEP_MS);
-            } catch (InterruptedException e) {
-                LOG.warn(e.getMessage());
-            }
+        if (appId == null) {
+            throw new LoadException(errMsg + "Failed to get appId from handle. spark app state: "
+                    + state.toString() + ", loadJobId:" + loadJobId);
         }
 
         // success
         attachment.setAppId(appId);
         attachment.setHandle(handle);
     }
 
-    public EtlStatus getEtlJobStatus(SparkAppHandle handle, String appId, long loadJobId, String etlOutputPath,
-                                     SparkResource resource, BrokerDesc brokerDesc) {
+    public EtlStatus getEtlJobStatus(SparkLoadAppHandle handle, String appId, long loadJobId, String etlOutputPath,
+                                     SparkResource resource, BrokerDesc brokerDesc) throws LoadException {
         EtlStatus status = new EtlStatus();
 
         if (resource.isYarnMaster()) {
-            // state from yarn
             Preconditions.checkState(appId != null && !appId.isEmpty());
```

Comment:
Maybe put statement `Preconditions.checkState ` outer the if statement?

Reply:
done